### ISSUE TITLE
Generated frameworks are empty

### ISSUE THREAD

**Issue by savage7** [grab/cocoapods-binary-cache#44]
### Checklist
- [x] Updated the plugin to the latest version
- [x] I've read the [Contribution Guidelines](https://github.com/grab/cocoapods-binary-cache/blob/master/CONTRIBUTING.md)
- [x] I've searched for [existing GitHub issues](https://github.com/grab/cocoapods-binary-cache/issues)

#### Command executed
I created a new repository for the cache, pushed one file so there is a master branch.
I modified my podfile (see the attached podfile).
Deleted all local folders regarding the plugin (_Prebuild, _Prebuild_delta, .cocoapods-binary-cache)
Called: 
bundle exec pod binary prebuild
bundle exec pod binary push

I attached my podfile. I dont have a dev pod. I hope you can help, your plugin would be a real time safer :) 

#### What went wrong?
xcodebuild seams to be called but all frameworks are empty. They are all 1kb big.
The framework zips are pushed to the repo, but are all empty.

#### Plugin version
The current latest commit:
https://github.com/grab/cocoapods-binary-cache/commit/bc9ebea42fbbcf0e147cb5a9b6750d3e58f8f4c2

#### Installed CocoaPods plugins
Cocoapods Version: 1.10.0

Plugins:
    - cocoapods-binary-cache : 0.1.11 (pre_install and post_install hooks)
    - cocoapods-deintegrate  : 1.0.4
    - cocoapods-plugins      : 1.0.0
    - cocoapods-search       : 1.0.0
    - cocoapods-trunk        : 1.5.0
    - cocoapods-try          : 1.2.0

[Podfile.txt](https://github.com/grab/cocoapods-binary-cache/files/5544436/Podfile.txt)

**Comment by trinhngocthuyen** [comment]
Hi @savage7, I tried running the plugin with the Podfile above and it seems working properly.
Please refer to that demo project here https://github.com/trinhngocthuyen/troubleshoot-pod-bin-cache/tree/1fca3aa/issue-44.

Kindly check it out and run the `run_me.sh` script. Then check the generated frameworks in `prebuilt-frameworks`. The prebuilt frameworks are actually in the correct structure.

**Comment by savage7** [comment]
@ivanrnld Can you please reopen this issue. Sorry I hadn't had the time to look into it into more detail before. 

I've now set up a debugging environment and was able to find the issue. However I'm not sure how the fix should look, especially because this works for everyone else.

First my environment in more detail:
Xcode 12.3 Build version 12C33
Cocoapods: 1.10.1 (also tried it with 1.10.1)
Ruby Version: 2.7.1 (also tried it with 2.6.0)

**Steps:**
- I've created a minimal Podfile only containing two pods
- FSCalendar is a binary pod SwiftyBeaver not binary
[Podfile.txt](https://github.com/grab/cocoapods-binary-cache/files/5819323/Podfile.txt)
- I delete the complete cache and all build artefacts before each test
- call: pod binary prebuild
- It successfully builds and the Framework is created in the build folder (build/Debug-iphonesimulator/FSCalendar/FSCalendar.framework)
- The plugin fails to copy the FSCalender.framework to _Prebuild/GeneratedFrameworks/FSCalendar

**Analysis:**
xcodebuilg_command.rb:run
path = target_products_dir_of(target, sdks[0]) + "/*"
results in:
/Users/removeduser/Downloads/debug_cocoapods_plugins_in_vscode-master/ios/build/Debug-iphonesimulator/FSCalendar/*

The problem is that xcode generate the folder like this:
![image](https://user-images.githubusercontent.com/971353/104701029-5829db80-5715-11eb-9769-42ea2c353cb9.png)

So the glob statement doesn't actually match FSCalendar.framework
It seams like you expect something like:
build/Debug-iphonesimulator/FSCalendar/FSCalendar.framework

When I adapt the statement to target_products_dir_of(target, sdks[0]) + "*" the Framework is matched and copied.
I've not created a pull request since this works for you differently I suspect another root cause for this problem. 

Can you please have another look into this?
I've setup a complete debugging environment, so if you need more debug output or if I should test some code don't hesitate to post it here :)

**Comment by trinhngocthuyen** [comment]
Hi @savage7, thanks for the effort in troubleshooting the issue.

Actually, the framework path of the target is determined by the **`CONFIGURATION_BUILD_DIR`** setting.
And CocoaPods set this value as `${PODS_CONFIGURATION_BUILD_DIR}/<PodName>` (in this case: `${PODS_CONFIGURATION_BUILD_DIR}/FSCalendar`).
That means the framework path should be `build/Debug-iphonesimulator/FSCalendar/FSCalendar.framework`)

Could you please help check this setting `CONFIGURATION_BUILD_DIR` whether it's correctly set?

**Comment by savage7** [comment]
Do you mean this:
[FSCalendar.release.xcconfig.txt](https://github.com/grab/cocoapods-binary-cache/files/5819758/FSCalendar.release.xcconfig.txt)

In the xcconfig CONFIGURATION_BUILD_DIR set like this:
CONFIGURATION_BUILD_DIR = ${PODS_CONFIGURATION_BUILD_DIR}/FSCalendar

**Comment by trinhngocthuyen** [comment]
The xcconfig can be overridden by the settings in the xcodeproj.
Is the value (under the target build settings of the given pod) displayed correctly in the project?
<img width="841" alt="Screenshot 2021-01-15 at 6 04 03 PM" src="https://user-images.githubusercontent.com/8844044/104711519-7109af80-575c-11eb-851c-5988d106355a.png">

**Comment by savage7** [comment]
Our projects are generated with **xodegen** and we don't set CONFIGURATION_BUILD_DIR manually.

**Without cocoapoda-binary-cache plugin:**
CONFIGURATION_BUILD_DIR=(removed this)build/Debug-iphonesimulator/FSCalendar

**With cocoapoda-binary-cache plugin:**
For FSCalendar:
CONFIGURATION_BUILD_DIR --> **is not set at all**, and other Cocoapods relevants vars too like PODS_BUILD_DIR
For SwiftyBeaver which is no binary pod:
CONFIGURATION_BUILD_DIR is set 
CONFIGURATION_BUILD_DIR=(removed this)build/Debug-iphonesimulator/SwiftyBeaver

So for binary pods CONFIGURATION_BUILD_DIR is not set, but for all other pods it is.

It seams for the binary Pod the configuration has not been set by cocoapods:
![image](https://user-images.githubusercontent.com/971353/104713685-7f3bd980-5724-11eb-84e6-5f24c7bb95de.png)

For most our project we have a setup where we use one target and this target has n configurations (like staging, production).

For a test I removed all the sub configurations and schemes:
**CONFIGURATION_BUILD_DIR is then set for FSCalendar correctly**
The Framework can be copied correctly and the cache looks fine

**Conclusion:**
Using different configurations like "staging debug/release" seams to cause this issue.

**Comment by savage7** [comment]
Update:
I updated the Troubleshooting project:
https://github.com/savage7/troubleshoot-pod-bin-cache/tree/master/issue-44

With this its reproducible.

I think I found out why this happens.
By Default xcodegen creates these configurations, there's no debug/release, when extra configs are created:
![image](https://user-images.githubusercontent.com/971353/104749161-c6d95a00-5752-11eb-8760-7aab7d79d243.png)

The plugins works when I manually define debug, release like this:
![image](https://user-images.githubusercontent.com/971353/104749241-dc4e8400-5752-11eb-9ae7-057e0ea3ab32.png)

I'm honestly not sure if thats an issue in the plugin, but without the plugins it just works.

**Comment by trinhngocthuyen** [comment]
I am afraid you're making a mistake in the configuration. As I can see from the attachments, you're using the `Production Debug` configuration in the project, but the prebuild config set as `Debug`
```rb
config_cocoapods_binary_cache(
  ...
  prebuild_config: "Debug"
)
```
As the `Debug` configuration is not set in the project, the build settings are improperly set up for this configuration, causing the issue.

To resolve, you need to either have a Debug configuration, or properly set the value of prebuild config:
```rb
config_cocoapods_binary_cache(
  ...
  prebuild_config: "'Production Debug'"
)
```

**Comment by savage7** [comment]
@trinhngocthuyen 
Yes I think basically it's a configuration error. But when I set 'Production Debug' as prebuild_config it's not found by xcodebuild for the Pods targets.

In fact setting "Debug" seams to be the correct choice. I changed my xcodegen configuration to also generate Debug. Which is in fact is the case in a "normal" xcode projects too. This works and the frameworks are not empty anymore.

One minimal change that could make sense, is to throw an error in the plugin when the configuration doesn't exist. 

Thanks alot for your help finding the root cause for this! You've created one of the coolest plugins for cocoapods ever :blush:
